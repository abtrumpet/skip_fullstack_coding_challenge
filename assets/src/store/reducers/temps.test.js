import { List } from "immutable";

import {
  TempsReducerRecord,
  tempsReducerInitialState,
  reducer,
} from "./temps";

describe("TempsReducerRecord", () => {
  test("it matches snapshot", () => {
    expect(new TempsReducerRecord()).toMatchSnapshot();
  });
});

describe("tempsReducerInitialState", () => {
  test("it matches snapshot", () => {
  
    expect(tempsReducerInitialState).toMatchSnapshot();
  });
});

describe("temps reducer", () => {
  test("it handles default on init", () => {
    expect(reducer(new TempsReducerRecord(), {})).toEqual(new TempsReducerRecord());
  });

  test("it handles default", () => {
    expect(reducer(new TempsReducerRecord({isFetching: true}), {})).toEqual(new TempsReducerRecord({isFetching: true}));
  });

  it("handles FETCH_TEMPS", () =>{
    expect(reducer(new TempsReducerRecord(), {type: "FETCH_TEMPS"})).toEqual(new TempsReducerRecord({isFetching: true}));
  });

  it("handles FETCH_TEMPS_SUCCESS", () => {
    expect(reducer(new TempsReducerRecord({isFetching: true}), {type: "FETCH_TEMPS_SUCCESS" })).toEqual(new TempsReducerRecord({isFetching: false}));
  });

  it("handle FETCH_TEMPS_FAILURE", () => {
    expect(reducer(new TempsReducerRecord({isFetching: true}), {type: "FETCH_TEMPS_FAILURE" })).toEqual(new TempsReducerRecord({isFetching: false}));
  });

  it("handles POST_TEMP", () => {
    expect(reducer(new TempsReducerRecord(), {type: "POST_TEMP"})).toEqual(new TempsReducerRecord({isPosting: true}));
  });

  it("handles POST_TEMP_SUCCESS", () => {
    expect(reducer(new TempsReducerRecord({ isPosting: true }), {type: "POST_TEMP_SUCCESS"})).toEqual(new TempsReducerRecord({isPosting: false}));
  });

  it("handles POST_TEMP_FAILURE", () => {
    expect(reducer(new TempsReducerRecord({ isPosting: true }), {type: "POST_TEMP_FAILURE"})).toEqual(new TempsReducerRecord({isPosting: false}));
  });

  it("handles ADD_TEMP 1", () => {
    expect(
      reducer(
        new TempsReducerRecord(), {type: "ADD_TEMP", temp: 10 }))
      .toEqual(new TempsReducerRecord({ fahrenheitTemps: List([10]) }));
  });

  it("handles ADD_TEMP 2", () => {
    expect(
      reducer(
        new TempsReducerRecord({ fahrenheitTemps: List([ 10 ])}), {type: "ADD_TEMP", temp: 30 }))
      .toEqual(new TempsReducerRecord({ fahrenheitTemps: List([ 10, 30 ]) }));
  });

  it("handles ADD_TEMP 3", () => {
    expect(
      reducer(
        new TempsReducerRecord({ fahrenheitTemps: List([ 10, 30, 50 ])}), {type: "ADD_TEMP", temp: 70 }))
      .toEqual(new TempsReducerRecord({ fahrenheitTemps: List([ 10, 30, 50, 70 ]) }));
  });

  const runTestSetTemps = ({ type, key }) => {
    it(`handles ${type} 1`, () => {
      expect(
        reducer(
          new TempsReducerRecord(), {type, temps: [1,2,3]}))
        .toEqual(new TempsReducerRecord({ [key]: List([1,2,3]) }));
    });

    it(`handles ${type} 2`, () => {
      expect(
        reducer(
          new TempsReducerRecord(), {type, temps: List([1,2,3])}))
        .toEqual(new TempsReducerRecord({ [key]: List([1,2,3]) }));
    });

    it(`handles ${type} 3`, () => {
      expect(
        reducer(
          new TempsReducerRecord(
            {
              [key]: List([1,2,3])
            }
          ), {type, temps: List([3,4,5])}))
        .toEqual(new TempsReducerRecord({ [key]: List([3,4,5]) }));
    });

    it(`handles ${type} 4`, () => {
      expect(
        reducer(
          new TempsReducerRecord(
            {
              [key]: List([1,2,3])
            }
          ), {type, temps: [3,4,5]}))
        .toEqual(new TempsReducerRecord({ [key]: List([3,4,5]) }));
    });
  };

  runTestSetTemps({ type: "SET_F_TEMPS", key: "fahrenheitTemps" });
  runTestSetTemps({ type: "SET_C_TEMPS", key: "celsiusTemps" });
  runTestSetTemps({ type: "SET_K_TEMPS", key: "kelvinTemps" });
});
