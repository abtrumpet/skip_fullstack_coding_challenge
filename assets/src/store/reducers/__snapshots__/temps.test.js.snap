// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TempsReducerRecord it matches snapshot 1`] = `
Immutable.Record {
  "fahrenheitTemps": Immutable.List [],
  "celsiusTemps": Immutable.List [],
  "kelvinTemps": Immutable.List [],
  "isFetching": false,
  "isPosting": false,
}
`;

exports[`tempsReducerInitialState it matches snapshot 1`] = `
Object {
  "celsiusTemps": Immutable.List [],
  "fahrenheitTemps": Immutable.List [],
  "isFetching": false,
  "isPosting": false,
  "kelvinTemps": Immutable.List [],
}
`;
